File: patch_healthbar.js
const fs = require('fs');
let code = fs.readFileSync('src/components/HealthBarItem.tsx', 'utf8');

code = code.replace(
  'readOnly?: boolean;',
  "readOnly?: boolean;\n  layout?: 'horizontal' | 'vertical';"
);

code = code.replace(
  'readOnly = false',
  "readOnly = false,\n  layout = 'horizontal'"
);

code = code.replace(
  'className={`bg-[#0c0d10] border border-bento-border rounded-xl p-4 ${readOnly ? \'\' : \'hover:border-slate-600\'} transition-all relative group`}',
  'className={`bg-[#0c0d10] border border-bento-border rounded-xl p-4 ${readOnly ? \'\' : \'hover:border-slate-600\'} transition-all relative group ${layout === \'vertical\' ? \'flex flex-col-reverse items-center justify-between h-[300px] w-[90px] shrink-0\' : \'\'}`}'
);

code = code.replace(
  '<div className={`flex justify-between items-center mb-2.5 relative z-10 transition-all duration-200 ${readOnly ? \'pr-2\' : \'pr-2 group-hover:pr-[220px] group-focus-within:pr-[220px]\'}`}>',
  '<div className={`flex justify-between items-center relative z-10 transition-all duration-200 ${layout === \'vertical\' ? \'flex-col gap-2 mb-0 mt-2\' : `mb-2.5 ${readOnly ? \'pr-2\' : \'pr-2 group-hover:pr-[220px] group-focus-within:pr-[220px]\'}`}`}>'
);

code = code.replace(
  '<div className="flex items-center gap-2 min-w-0">',
  '<div className={`flex items-center gap-2 min-w-0 ${layout === \'vertical\' ? \'flex-col\' : \'\'}`}>'
);

code = code.replace(
  '<span className="font-display font-bold text-slate-200 tracking-wide text-sm md:text-base truncate">',
  '<span className={`font-display font-bold text-slate-200 tracking-wide text-sm md:text-base ${layout === \'vertical\' ? \'break-all text-center leading-tight [writing-mode:vertical-rl] rotate-180\' : \'truncate\'}`}>'
);

code = code.replace(
  '<div className="font-mono text-xs text-slate-400 shrink-0">',
  '<div className={`font-mono text-xs text-slate-400 shrink-0 flex items-center ${layout === \'vertical\' ? \'flex-col\' : \'\'}`}>'
);

code = code.replace(
  '<div className="relative">',
  '<div className={`relative ${layout === \'vertical\' ? \'flex-grow w-full flex justify-center\' : \'\'}`}>'
);

code = code.replace(
  'className={`flex h-8 w-full rounded-lg bg-[#1a1d23] overflow-hidden border border-[#2d333d]',
  'className={`flex ${layout === \'vertical\' ? \'flex-col-reverse w-8 h-full\' : \'h-8 w-full\'} rounded-lg bg-[#1a1d23] overflow-hidden border border-[#2d333d]'
);

code = code.replace(
  'className="h-full flex-grow rounded-[2px] transition-all duration-200"',
  'className={`${layout === \'vertical\' ? \'w-full\' : \'h-full\'} flex-grow rounded-[2px] transition-all duration-200`}'
);


fs.writeFileSync('src/components/HealthBarItem.tsx', code);

File: patch_healthbar.py
import sys

with open('src/components/HealthBarItem.tsx', 'r') as f:
    code = f.read()

# 1. Add barContainerRef
code = code.replace("const prevValueRef = useRef(bar.currentValue);", "const prevValueRef = useRef(bar.currentValue);\n  const barContainerRef = useRef<HTMLDivElement>(null);\n\n  useEffect(() => {\n    if (!isMouseDown || activeBarIdRef.current !== bar.id || readOnly) return;\n\n    const handleMouseMove = (e: MouseEvent) => {\n      if (!barContainerRef.current) return;\n      const rect = barContainerRef.current.getBoundingClientRect();\n      let newPercentage = 0;\n      if (layout === 'vertical') {\n        const y = e.clientY - rect.top;\n        newPercentage = 1 - (y / rect.height);\n      } else {\n        const x = e.clientX - rect.left;\n        newPercentage = x / rect.width;\n      }\n      newPercentage = Math.max(0, Math.min(1, newPercentage));\n      const newValue = Math.round(newPercentage * bar.maxValue);\n      if (newValue !== bar.currentValue) {\n        handleSegmentInteraction(bar, newValue);\n      }\n    };\n\n    window.addEventListener('mousemove', handleMouseMove);\n    return () => window.removeEventListener('mousemove', handleMouseMove);\n  }, [isMouseDown, bar, layout, readOnly, handleSegmentInteraction, activeBarIdRef]);")


# 2. Update the vertical bar container
code = code.replace("className={`flex flex-col-reverse w-full h-full rounded bg-[#1a1d23] overflow-hidden border border-[#2d333d]", "ref={barContainerRef}\n                className={`flex flex-col-reverse w-full h-full rounded bg-[#1a1d23] overflow-hidden border border-[#2d333d]")

# 3. Update the horizontal bar container
code = code.replace("className={`flex h-8 w-full rounded-lg bg-[#1a1d23] overflow-hidden border border-[#2d333d]", "ref={barContainerRef}\n            className={`flex h-8 w-full rounded-lg bg-[#1a1d23] overflow-hidden border border-[#2d333d]")


with open('src/components/HealthBarItem.tsx', 'w') as f:
    f.write(code)

